test/extended/operators/certs.go

```diff
@@ -49,6 +56,82 @@ var _ = g.Describe("[sig-arch][Late]", func() {
+		o.Expect(err).NotTo(o.HaveOccurred())
+		for i, node := range nodeList.Items {
+			nodes[node.Name] = i
+		}
+
+		actualPKIContent, err := certgraphanalysis.GatherCertsFromPlatformNamespaces(ctx, kubeClient)
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		expectedPKIContent, err := certs.GetPKIInfoFromEmbeddedOwnership()
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		messages := []string{}
+
+		unregisteredSecrets := map[certgraphapi.InClusterSecretLocation]certgraphapi.PKIRegistryInClusterCertKeyPair{}
+
+		for _, currCertKeyPair := range actualPKIContent.InClusterResourceData.CertKeyPairs {
+			currLocation := currCertKeyPair.SecretLocation
+			expectedSecret, err := certgraphutils.LocateCertKeyPair(currLocation, expectedPKIContent.CertKeyPairs)
+			if err != nil {
+				unregisteredSecrets[currLocation] = currCertKeyPair
+				continue
+			}
+			if expectedSecret.CertKeyInfo.Description == "" || expectedSecret.CertKeyInfo.OwningJiraComponent == "" {
+				continue
+			}
+			errorMessage := o.Equal(expectedSecret.CertKeyInfo).FailureMessage(currCertKeyPair.CertKeyInfo)
+			if len(errorMessage) > 0 {
+				messages = append(messages, fmt.Sprintf("--namespace=%s, secret/%s:\n%v\n", currLocation.Namespace, currLocation.Name, errorMessage))
+			}
+		}
+		if len(unregisteredSecrets) > 0 {
+			for _, secret := range unregisteredSecrets {
+				messages = append(messages, fmt.Sprintf("Unregistered TLS artifact: --namespace=%s, secret/%s\n", secret.SecretLocation.Namespace, secret.SecretLocation.Name))
+			}
+		}
+
+		unregisteredConfigMaps := map[certgraphapi.InClusterConfigMapLocation]certgraphapi.PKIRegistryInClusterCABundle{}
+		for _, currCABundle := range actualPKIContent.InClusterResourceData.CertificateAuthorityBundles {
+			currLocation := currCABundle.ConfigMapLocation
+			expectedCABundle, err := certgraphutils.LocateCertificateAuthorityBundle(currLocation, expectedPKIContent.CertificateAuthorityBundles)
+			if err != nil {
+				unregisteredConfigMaps[currLocation] = currCABundle
+				continue
+			}
+			if expectedCABundle.CABundleInfo.Description == "" || expectedCABundle.CABundleInfo.OwningJiraComponent == "" {
+				continue
+			}
+			errorMessage := o.Equal(expectedCABundle.CABundleInfo).FailureMessage(currCABundle.CABundleInfo)
+			if len(errorMessage) > 0 {
+				messages = append(messages, fmt.Sprintf("--namespace=%s, configmap/%s:\n%v\n", currLocation.Namespace, currLocation.Name, errorMessage))
+			}
+		}
+
+		if len(unregisteredConfigMaps) > 0 {
+			for _, cm := range unregisteredConfigMaps {
+				messages = append(messages, fmt.Sprintf("Unregistered TLS artifact: --namespace=%s, configmap/%s\n", cm.ConfigMapLocation.Namespace, cm.ConfigMapLocation.Name))
+			}
+		}
+
+		if len(messages) > 0 {
+			// TODO: uncomment when test no longer fails and enhancement is merged
+			//g.Fail(strings.Join(messages, "\n"))
+			testresult.Flakef(strings.Join(messages, "\n"))
```

Comment:
I'd rather keep it in flake longer - for about a week, so that all payload blocking / periodic jobs would run at least once and all existing secrets are registered.